core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -195,327 +215,176 @@ class WorkspaceService(
-                )
-              } else {
-                noFuture
-              }
-
-              def ownersFuture(): Future[Option[Set[LookupExpression]]] = if (options.contains("owners")) {
-                traceFutureWithParent("getWorkspaceOwners", s1)(_ =>
-                  getWorkspaceOwners(workspaceContext.workspaceIdAsUUID.toString).map(_.map(_.value)).map(Option(_))
-                )
-              } else {
-                noFuture
-              }
+  def getWorkspace(workspaceName: WorkspaceName, params: WorkspaceFieldSpecs): Future[JsObject] = {
+    val options = processOptions(params)
+    traceFutureWithParent("getV2WorkspaceContextAndPermissions", ctx)(_ =>
+      for {
+        workspace <-
+          getV2WorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.read, Option(options.attrSpecs))
+        workspaceResponse <- getWorkspaceDetails(workspace, options)
+      } yield
+      // post-process JSON to remove calculated-but-undesired keys
+      deepFilterJsObject(workspaceResponse.toJson.asJsObject, options.options)
+    )
+  }
 
-              def workspaceAuthorizationDomainFuture(): Future[Option[Set[ManagedGroupRef]]] =
-                if (options.contains("workspace.authorizationDomain") || options.contains("workspace")) {
-                  traceFutureWithParent("loadResourceAuthDomain", s1)(_ =>
-                    loadResourceAuthDomain(SamResourceTypeNames.workspace, workspaceContext.workspaceId, ctx.userInfo)
-                      .map(Option(_))
-                  )
-                } else {
-                  noFuture
-                }
+  def getWorkspaceById(workspaceId: String, params: WorkspaceFieldSpecs): Future[JsObject] = {
+    val options = processOptions(params)
+    traceFutureWithParent("getV2WorkspaceContextAndPermissions", ctx)(_ =>
+      for {
+        workspace <-
+          getV2WorkspaceContextAndPermissionsById(workspaceId, SamWorkspaceActions.read, Option(options.attrSpecs))
+        workspaceResponse <- getWorkspaceDetails(workspace, options)
+      } yield
+      // post-process JSON to remove calculated-but-undesired keys
+      deepFilterJsObject(workspaceResponse.toJson.asJsObject, options.options)
+    )
+  }
 
-              def workspaceSubmissionStatsFuture(): ReadAction[Option[WorkspaceSubmissionStats]] =
-                if (options.contains("workspaceSubmissionStats")) {
-                  getWorkspaceSubmissionStats(workspaceContext, dataAccess).map(Option(_))
-                } else {
-                  DBIO.from(noFuture)
-                }
+  def getWorkspaceDetails(workspace: Workspace, options: QueryOptions): Future[WorkspaceResponse] = {
+    val workspaceId = workspace.workspaceId
+    /*
+    If we're looking to improve performance, we could potentially use this, instead trying to run futures in parallel:
+    userRoles <- samDAO.listUserActionsForResource(SamResourceTypeNames.workspace, workspaceId, ctx)
+     */
```

Comment:
This is just referring to removing the futuresInParallel usage. I suspect `getWorkspace` with the parallel futures was all done before that batch endpoint was added.
It doesn't seem particularly important anyway, since this is all only ever for a single workspace, some calls couldn't be run in parallel anyway.

It was also very difficult to follow, since it was all nested inside two other futures (for `wsmService.fetchAggregatedWorkspace` and `accessLevelFuture()`), which was all wrapped in a slick transaction.